src/plugins/intel_npu/src/al/include/intel_npu/al/icompiler.hpp

```diff
@@ -135,6 +138,7 @@ struct NetworkDescription final {
     NetworkDescription(std::vector<uint8_t>&& compiledNetwork, NetworkMetadata&& metadata)
         : compiledNetwork(std::move(compiledNetwork)),
           metadata(std::move(metadata)) {}
+    NetworkDescription(NetworkMetadata&& metadata) : metadata(std::move(metadata)) {}
```

Comment:
We should not add such conversion. This creates "empty" NetworkDescription instance which is not representing the compiled model, breaking the semantics. With this change I have two types of ND to deal with making code harder to reason about,.
Instead let's pass around NetworkMetadata and only create ND using prior ctor once we have both elements at hand.

Reply:
is that ok if we with revert this change
and perform `auto networkDescription = NetworkDescription(std::vector<uint8_t>(), std::move(networkMeta));` in compile()
in this case, we can remain and only have 1 constructor, but we passing an empty blob to it

Reply:
I may not agree with you on this but not sure that I understand correctly what you are saying here. What exactly do you mean by "empty" network description? By its name, NetworkDescription, I don't expect to keep a blob inside of it, but only a description as metadata.

Maybe is better to keep it this way and just get the blob if needed through the getCompiledNetwork method if needed(in the case of CiP). This would mean extra changes on the CiP path.

Reply:
The compiledModel is thing we want reduce with this PR, I think ND is a wrap of original network, we do not need to bind it to compiledModel.
For CiP path, release the compiledModel after pfnCreate in zeroBackend shall also save memory usage,  so we need NetworkDescription with empty CompiledModel once we consumed the CompiledMode.